mlir/lib/Catalyst/Driver/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+if(WIN32)
+    set(EXTERNAL_LIB Version)
+elseif(UNIX AND NOT APPLE)
+    set(EXTERNAL_LIB z tinfo)
+elseif(APPLE)
+    set(EXTERNAL_LIB z ncurses)
+endif()
+
+set(ENZYME_STATIC_LIB ON)
+add_subdirectory(${ENZYME_SRC_DIR}/enzyme ${CMAKE_CURRENT_BINARY_DIR}/enzyme)
+
+include_directories(${ENZYME_SRC_DIR}/enzyme/Enzyme)
+
+# Taken from llvm/tools/llc/CMakeLists.txt
+# TODO: May be possible to simplify linked libs
+set(LLVM_LINK_COMPONENTS
```

Comment:
I was trying to find a good documentation / source for `LLVM_LINK_COMPONENTS` but I was not able to find one fast. It is a common variable used across the LLVM building infrastructure for libraries. I know it will be used in `add_mlir_library`.